### PR TITLE
complete rewrite, update CMakeLists.txt to 3.5 and change fPIC

### PR DESCRIPTION
**Pull request by AlwinEsch**
As a result of this addon no longer working under GL, I have the GL code completely cleaned up.

I tested this under Linux, Mac and Windows (Angle), where it ran normally.

In addition, the GL shaders have been moved to their own folder, and they are back in their original form (without DX hacks).

In order to continue to support DX, the used ones are still in the old folder (becomes removed in Matrix).

**In addition, the new code is ready for use on Windows via Angle and ran perfect during the test (need only Angle in depends and bit changed CMakelists.txt).**

### CODE REVIEW

**Comment by AlwinEsch** [comment]
Ping; from the CMakeLists.txt update goes it now to a big change :wink:

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,62 +1,44 @@
...
-    if(OPENGLES2_FOUND)
-      set(INCLUDES ${OPENGLES2_INCLUDE_DIR})
-      add_definitions(-DHAS_GLES2 -DHAS_GLES ${OpenGLES2_CFLAGS_OTHER})
-      set(DEPLIBS ${DEPLIBS} ${OPENGLES2_LIBRARIES})
-      set(APP_RENDER_SYSTEM gles)
+  include_directories(${GLM_INCLUDE_DIR}
+                      ${KODI_INCLUDE_DIR}/.. # Hack way with "/..", need bigger Kodi cmake rework to match right include ways
+                      ${PROJECT_SOURCE_DIR}/lib)
+
+  if(NOT ${CORE_SYSTEM_NAME} STREQUAL "")
```
Comment (Rechi) [inline CMakeLists.txt]: Not needed, the bellow if already implies that `CORE_SYSTEM_NAME` isn't empty.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,62 +1,44 @@
...
+    if(${CORE_SYSTEM_NAME} STREQUAL "osx" OR
+       ${CORE_SYSTEM_NAME} STREQUAL "ios" OR
+       ${CORE_SYSTEM_NAME} STREQUAL "darwin_embedded")
+      list(APPEND DEPLIBS "-framework CoreVideo")
     endif()
   endif()
-  if(NOT APP_RENDER_SYSTEM)
-    message(FATAL_ERROR "No rendering API found. Bailing")
```
Comment (Rechi) [inline CMakeLists.txt:41]: Should be in the else branch of the above if `APP_RENDER_SYSTEM`

File: FindOpenGLES.cmake [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+# OPENGLES_INCLUDE_DIRS - the OpenGLES include directory
+# OPENGLES_LIBRARIES - the OpenGLES libraries
+# OPENGLES_DEFINITIONS - the OpenGLES definitions
+#
+# Note:
+# On Windows with angle the *_INCLUDE_DIRS and
+# *_DEFINITIONS are undefined, but are set
+# global by the kodi-angle package.
+
+if(WIN32)
```
Comment (Rechi) [inline FindOpenGLES.cmake]: not used on windows, so drop everything related to it

Comment (AlwinEsch) [reply FindOpenGLES.cmake]: Thought to prepare direct for Matrix, becomes removed.

File: CMakeLists.txt (hunk 2 above)
Comment (AlwinEsch) [reply CMakeLists.txt:41]: This is not required, thats why removed.
cmake's find_package generates following if not present:
```
CMake Error at /usr/share/cmake-3.13/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find OpenGl (missing: OPENGL_gl_LIBRARY OPENGL_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake-3.13/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  FindOpenGl.cmake:31 (find_package_handle_standard_args)
  CMakeLists.txt:13 (find_package)
```

EDIT:
Also must, if used, it changed to:
```cmake
  if(NOT OPENGL_FOUND AND NOT OPENGLES_FOUND)
    message(FATAL_ERROR "No rendering API found. Bailing")
  endif()
```
Still possible that `APP_RENDER_SYSTEM` is not set from Kodi.

Comment (Rechi) [reply CMakeLists.txt:41]: There is still a case left, `APP_RENDER_SYSTEM` not empty, not gl and not gles.
Either add an else branch or replace `elseif(APP_RENDER_SYSTEM STREQUAL "gles")` with `else()`.

Comment (AlwinEsch) [reply CMakeLists.txt:41]: I change to `else()`.

As question, which render system except "GL" and "GLES" could still be defined?
If anything else came along, there would be something bigger about Kodi and the build system broken.

**Comment by AlwinEsch** [comment]
Is updated

File: FindOpenGLES.cmake [hunk 4]
```diff
@@ -0,0 +1,60 @@
+#.rst:
+# FindOpenGLES
+# ------------
+# Finds the OpenGLES2 library
```
Comment (Rechi) [inline FindOpenGLES.cmake]: finds OpenGLES2 and OpenGLES3

File: FindOpenGLES.cmake [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+#
+# OPENGLES_FOUND - system has OpenGLES
+# OPENGLES_INCLUDE_DIRS - the OpenGLES include directory
+# OPENGLES_LIBRARIES - the OpenGLES libraries
+# OPENGLES_DEFINITIONS - the OpenGLES definitions
+#
+# Note:
+# On Windows with angle the *_INCLUDE_DIRS and
+# *_DEFINITIONS are undefined, but are set
+# global by the kodi-angle package.
```
Comment (Rechi) [inline FindOpenGLES.cmake]: What's that?

File: FindOpenGLES.cmake (hunk 4 above)
Comment (AlwinEsch) [reply FindOpenGLES.cmake]: A remnant where that came from Kodi: https://github.com/xbmc/xbmc/blob/master/cmake/modules/FindOpenGLES.cmake#L4

File: FindOpenGLES.cmake (hunk 5 above)
Comment (AlwinEsch) [reply FindOpenGLES.cmake]: No comment, only so much, a singing Chipmunk is often not in his senses. :innocent:

**Pull request merged**